### DISCUSSION TITLE
Gitlab as Storage in Settings - Unclear how to set up

### DISCUSSION THREAD

**Discussion by ambedded** [requarks/wiki#7074] [General]
Hi

Im really unclear with the options in under storage/git. Its unclear how to set up git(lab) with ssh and a private key. Can you please make it clear which options are required, which one are optional and provide clear and easy to understand text-assistance for each option?

as of now, latest version of wikijs, its a guessing game, try and error and after 60min i gave up.

would be really cool to make this feature available, as for now, i consider it "non existing" since i cannot get it to work.

**Comment by fnafox** [comment]
Git storage options are fairly clear, I don't see which part is a guessing game as you say.
Having extra fields filled out won't make a difference as it uses only the ones it needs.
- Select the authentication type, ssh since you want to use ssh.
- Enter the git URI git@gitlab.com:org/repo.git
- Enter branch name
- Select whether you want to write the path to key file or paste the contents directly
- Either enter the path to your key file or paste the private key contents in the appropriate box
- Enter default author email and name
- Set a local repo path
- Select sync direction and schedule
- Enable and save your storage target
